### ISSUE TITLE
ARM build

### ISSUE THREAD

**Issue by bfosterjr** [PJayB/DOOM-3-BFG-DX11#1]
Have you looked at an ARM build for this?

**Comment by PJayB** [comment]
Not yet. It wasn't too difficult for the Q3 port. I don't really have the bandwidth to investigate this right now, but I encourage you to give it a shot :)

**Comment by bfosterjr** [comment]
Thanks for the response.

I was asking because I am looking at this right now. I've hacked away all the SSE2 / assembly code. I've gotten everything to compile cleanly with the exception of 'neo\d3xp\brittlefracture.cpp' which cl bails on with a C1002 error. Its very bizarre. I've even tried with the x64 arm cross tools.. same result.. bails after trying to use >7GB of memory. This is using VS2013 (update5) with 8.1 WDK.

Anyway. if you have any quick thoughts on this.. I'd appreciate hearing them. If not.. no worries.. I'll keep hammering away on it.

Cheers!

**Comment by PJayB** [comment]
Hmm, strange! Usually the solution to C1002 is to break the source file up,
but it isn't particularly long to start with...

Can you provide source? Maybe VS2015 would fare better. Plus, I can try and
repro the C1002 on my machine in VS2013.

**Comment by bfosterjr** [comment]
I've narrowed it down to an optimization bug. If i simply disable optimization on that one file.. I can get a complete build. Now to work on the DirectX8 input issues.. :(

**Comment by bfosterjr** [comment]
Well. I thought I had it until I realized that shader model 5.0 won't work on Windows RT (surface 2). It needs to be 4.0 (level 9_1). Sadly I'm not a game dev and I've exhausted my knowledge here as there is some code that isn't compatible and I don't have any confidence in how to proceed.

eg:

6>C:\code\DOOM-3-BFG-DX11-master\neo\renderer\hlsl\global.inc(118,12-19): error X4507: maximum vs_4_0_level_9_1 constant register index (256) exceeded - Try reducing number of constants referenced

I figure i can expand this to two arrays.. but i'm not sure what the implies for the rest of the code..

then there is also this:

1>C:\code\DOOM-3-BFG-DX11-master\neo\renderer\hlsl\heathaze.pixel(47,18-25): error X4502: Shader model ps_4_0_level_9_1 doesn't allow reading from position semantics.
